src/cargo/ops/cargo_install.rs

```diff
@@ -55,37 +56,75 @@ impl Drop for Transaction {
 }
 
 pub fn install(root: Option<&str>,
+               krates: Vec<&str>,
+               source_id: &SourceId,
+               vers: Option<&str>,
+               opts: &ops::CompileOptions,
+               force: bool) -> CargoResult<()> {
+    let root = resolve_root(root, opts.config)?;
+    let map = SourceConfigMap::new(opts.config)?;
+
+    if krates.len() <= 1 {
+        install_one(root, map, krates.into_iter().next(), source_id, vers, opts, force)
+    } else {
+        let mut success = vec![];
+        let mut errors = vec![];
+        for krate in krates {
+            let root = root.clone();
+            let map = map.clone();
+            match install_one(root, map, Some(krate), source_id, vers, opts, force) {
+                Ok(()) => success.push(krate),
+                Err(e) => errors.push(format!("{}: {}", krate, e))
```

Comment:
Basically I didn't want errors to be lost in a sea of command-line output, so I thought it'd be better to collect them at the end.